modules/core/src/lib/attribute-manager.js

```diff
@@ -224,6 +224,8 @@ export default class AttributeManager {
     // keep track of whether some attributes are updated
     let updated = false;
 
+    const externalAttributes = this._updateExternalAttributes(data);
```

Comment:
It seems like a lot to add a whole new concept of "external attributes" to make the API slightly prettier.

Reply:
A little background: we are trying to address multiple use cases here

- Support loaders.gl output format (good to have)
- Provide an official API for users to provide their own buffers to layers. This is currently supported (but undocumented) by checking attribute names in the layer props.
  + Users have to look into the layer code to understand what attributes there are. The "attribute alias" feature added in this PR is trying to relax this constraint.
  + The existing feature only accepts typed arrays, which means that users have to pack the external buffer exactly the same as that by the layer itself. For example, an external buffer for position that is in XYXYXY instead of XYZXYZXYZ cannot be used, because there is no way to set the stride.

The second use case is much bigger, and fundamental to our future GPGPU work.